src/dos/dos_programs.cpp

```diff
@@ -327,31 +327,8 @@ class MOUNT : public Program {
 				int num = -1;
 				cmd->FindInt("-usecd",num,true);
 				int error = 0;
-				if (cmd->FindExist("-ioctl_dio", false)) {
-					MSCDEX_SetCDInterface(CDROM_USE_IOCTL_DIO, num);
-				} else if (cmd->FindExist("-ioctl_dx",false)) {
-					MSCDEX_SetCDInterface(CDROM_USE_IOCTL_DX, num);
-#if defined (WIN32)
-				} else if (cmd->FindExist("-ioctl_mci",false)) {
-					MSCDEX_SetCDInterface(CDROM_USE_IOCTL_MCI, num);
-#endif
-				} else if (cmd->FindExist("-noioctl",false)) {
+				if (cmd->FindExist("-noioctl",false)) {
```

Comment:
Previously under Windows a default `mount d f:\ -t cdrom` would give them an `-ioctl`-type mount with an MSCDEX interface.
With `-ioctl` now gone under Windows, what happens to `mount d f:\ -t cdrom`?  I think we have two options:
1. Quit with an error and inform the user that implied `-ioctl` mounts are not available, and they must specify `-noioctl` from now on.
1. Switch and make `-noioctl` the default, so they still get an MSCDEX interface via MSCDEX_SetCDInterface.

Reply:
Huh, good point; when making this change few days ago I was *convinced*, that all code paths route back to SDL as a fallback in the end, but I'm not 100% sure any more - I'll double check.

If it'll turn out to be true and -ioctl/-noioctl is pointless now on Windows (same way as it always was pointless on macOS), then I'll just print some user warnings.

Overall, this will be a bigger deal when removing last remains of physical support - I marked some places in code, that will need to trigger warnings/errors to the users trying to mount

Reply:
OK, I double checked - when `-t cdrom`  is passed, then no matter if `-ioctl`, `-noioctl`, or nothing at all is passed - each time SDL CD-ROM interface will be selected (except via slightly different codepaths). If destination is not a real cdrom, then it will fallback to mounting as directory (as if `-t cdrom` was not passed at all) - I don't like this behaviour, but perhaps it's the best for backwards-compatibility.

Also, on Linux parameters `-ioctl`/`-noioctl` do not make sense, as both fall into ioctl interface inheriting from SDL interface.

Added a warning to notify user, that `-ioctl` does not work any more.